src/qibocal/protocols/characterization/signal_experiments/time_of_flight_readout.py

```diff
@@ -0,0 +1,185 @@
+from dataclasses import dataclass, field
+from typing import Optional
+
+import numpy as np
+import numpy.typing as npt
+import plotly.graph_objects as go
+from qibolab import AcquisitionType, AveragingMode, ExecutionParameters
+from qibolab.platform import Platform
+from qibolab.pulses import PulseSequence
+from qibolab.qubits import QubitId
+
+from qibocal.auto.operation import Data, Parameters, Qubits, Results, Routine
+
+
+@dataclass
+class TimeOfFlightReadoutParameters(Parameters):
+    """TimeOfFlightReadout runcard inputs."""
+
+    nshots: Optional[int] = None
+    """Number of shots."""
+    relaxation_time: Optional[int] = None
+    """Relaxation time (ns)."""
+    window_size: Optional[int] = 40
+    """Window size for the moving average."""
+
+
+@dataclass
+class TimeOfFlightReadoutResults(Results):
+    """TimeOfFlightReadout outputs."""
+
+    fitted_parameters: dict[QubitId, dict[str, float]]
+    """Raw fitting output."""
+
+
+TimeOfFlightReadoutType = np.dtype([("samples", np.float64), ("window_size", np.int64)])
+
+
+@dataclass
+class TimeOfFlightReadoutData(Data):
+    """TimeOfFlightReadout acquisition outputs."""
+
+    data: dict[QubitId, npt.NDArray[TimeOfFlightReadoutType]] = field(
+        default_factory=dict
+    )
+    """Raw data acquired."""
+
+    def register_qubit(self, qubit, samples, window_size):
+        """Store output for single qubit."""
+        shape = (1,) if np.isscalar(samples) else samples.shape
+        ar = np.empty(shape, dtype=TimeOfFlightReadoutType)
```

Comment:
```suggestion
        ar = np.empty(samples.shape, dtype=TimeOfFlightReadoutType)
```
Since you are appending data only once at the end you will never have samples as a scalar, right?

Reply:
Yup